src/NServiceBus.Hosting.Windows/StartableAndStoppable/StartableAndStoppableRunner.cs

```diff
@@ -96,8 +109,30 @@ public Task Start(IMessageSession session)
             }
         }
 
+        CancellationTokenSource LongRunningWarning(string message)
+        {
+            var tokenSource = new CancellationTokenSource();
+
+            Task.Run(() =>
+            {
+                try
+                {
+                    Task.Delay(LongRunningWarningTimeSpan).Wait(tokenSource.Token);
```

Comment:
Even with that conditional logging bit, there is no reason to be using `Wait` here since you can just use an async lambda.